### PR TITLE
Add SR850.yaml

### PR DESCRIPTION
**Pull request by muratbulat**
Reference: https://lenovopress.lenovo.com/lp1052.pdf

### CODE REVIEW

**Review by muratbulat — COMMENTED** [review]
ThinkSystem SR850

**Review by muratbulat — COMMENTED** [review]
lower

File: device-types/Lenovo/SR850.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+part_number: 7B27A03953
+u_height: 2
+is_full_depth: true
+comments: '[Lenovo System SR850 product guide](https://lenovopress.lenovo.com/lp1052.pdf)'
+subdevice_role: parent
+interfaces:
+  - name: XCC
+    type: 1000base-t
+    mgmt_only: true
+power-ports:
```
Comment (danner26) [inline device-types/Lenovo/SR850.yaml:14]: According to the datasheet, the PSUs are hot-swappable. So the power slots should be created in `module-bays`, and then the associated PSU's themselves should be added to the module-types folder if they do not already exist

File: device-types/Lenovo/SR850.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+manufacturer: Lenovo
+model: ThinkSystem SR850
+slug: sr850
+part_number: 7B27A03953
+u_height: 2
+is_full_depth: true
+comments: '[Lenovo System SR850 product guide](https://lenovopress.lenovo.com/lp1052.pdf)'
+subdevice_role: parent
+interfaces:
+  - name: XCC
```
Comment (danner26) [inline device-types/Lenovo/SR850.yaml:11]: Is there only one interface on this device?

File: device-types/Lenovo/SR850.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+manufacturer: Lenovo
+model: ThinkSystem SR850
+slug: sr850
+part_number: 7B27A03953
+u_height: 2
+is_full_depth: true
+comments: '[Lenovo System SR850 product guide](https://lenovopress.lenovo.com/lp1052.pdf)'
+subdevice_role: parent
```
Comment (danner26) [inline device-types/Lenovo/SR850.yaml:9]: `subdevice_role: parent` should only be used when there are children devices, which would be housed under `device-bays`

An example of this is the Dell VRTX chassis. The chassis itself is a device, which is a parent to blades such as the M620.

Unless this device houses other devices, which I cannot find information saying it does, then please remove this definition.

File: device-types/Lenovo/SR850.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+manufacturer: Lenovo
+model: ThinkSystem SR850
+slug: sr850
```
Comment (danner26) [inline device-types/Lenovo/SR850.yaml:4]: The slug for this should be `lenovo-thinksystem-sr850`

File: device-types/Lenovo/SR850.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+part_number: 7B27A03953
+u_height: 2
+is_full_depth: true
+comments: '[Lenovo System SR850 product guide](https://lenovopress.lenovo.com/lp1052.pdf)'
```
Comment (danner26) [inline device-types/Lenovo/SR850.yaml:8]: Please add `weight` and `airflow` definitions

File: device-types/Lenovo/SR850.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+---
+manufacturer: Lenovo
+model: ThinkSystem SR850
+slug: sr850
+part_number: 7B27A03953
```
Comment (danner26) [inline device-types/Lenovo/SR850.yaml:5]: This part number appears to only be for the processor and memory expansion tray. This is not the part number for the device itself.

**Review by danner26 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**